parser.go

```diff
@@ -1072,7 +1124,16 @@ func (p *parser) getSchemaObjectCached(pkgPath, pkgName, typeName string) (*Sche
 	var schemaObject *SchemaObject
 
 	// see if we've already parsed this type
-	if knownObj, ok := p.KnownIDSchema[genSchemeaObjectID(pkgName, typeName)]; ok {
+	knownObj := checkCache(pkgName, typeName, p)
+	if knownObj != nil {
+		schemaObject = knownObj
+	}
+
+	if schemaObject != nil {
```

Comment:
This second check seems unnecessary. Under what circumstance would `schemaObject` not be `nil` if it hadn't been assigned in the previous if statement? Would it be easier to just do

```
if knownObj != nil {
  return knownObj, nil
}

var schemaObject *SchemaObject
... etc
```